### PR TITLE
implemented "create" property on GCSFS and opener

### LINKED ISSUE #8: GCFS can be initialised in non existing folder
```Python
from fs import open_fs
open_fs("gs://valid_bucket_name/directory_that_does_not_exist")
```
completes without error, even though the directory does not exist

### PR DESCRIPTION
**Pull request by birnbaum**
Fixes #8

### CODE REVIEW

File: fs_gcsfs/_gcsfs.py [hunk 1]
```diff
@@ -66,8 +71,20 @@ def __init__(self,
...
+            raise CreateFailed("The provided bucket does not seem to exist") from err
+        except google.api_core.exceptions.Forbidden as err:
+            raise CreateFailed("You don't have access to the provided bucket") from err
+
+        if create:
+            root_marker = self._get_blob(forcedir(root_path))
+            if not root_marker:
+                blob = self.bucket.blob(forcedir(root_path))
+                blob.upload_from_string(b"")
+        elif strict and not self._get_blob(forcedir(root_path)):
```
Comment (trendelkampschroer) [inline fs_gcsfs/_gcsfs.py]: I think `and self._get_blob(...) is None`  should be preferred here.

File: fs_gcsfs/_gcsfs.py [hunk 2]
```diff
@@ -66,8 +71,20 @@ def __init__(self,
...
+        try:
+            self.bucket = self.client.get_bucket(self._bucket_name)
+        except google.api_core.exceptions.NotFound as err:
+            raise CreateFailed("The provided bucket does not seem to exist") from err
+        except google.api_core.exceptions.Forbidden as err:
+            raise CreateFailed("You don't have access to the provided bucket") from err
+
+        if create:
+            root_marker = self._get_blob(forcedir(root_path))
+            if not root_marker:
```
Comment (trendelkampschroer) [inline fs_gcsfs/_gcsfs.py]: I think `if root_marker is None` should be preferred here.

File: fs_gcsfs/_gcsfs.py [hunk 3]
```diff
@@ -66,8 +71,20 @@ def __init__(self,
...
+                blob = self.bucket.blob(forcedir(root_path))
```
Comment (trendelkampschroer) [inline fs_gcsfs/_gcsfs.py]: I don't remember if that's possible but maybe `self.makedir(root_path)` could be used instead?

Comment (birnbaum) [reply fs_gcsfs/_gcsfs.py]: this was my first take too, but `makedir` already uses all the internal GCSFS functions that already operate on `root_path`.

So for `root_path="some/path"`, `makedir("some/path")` would create `some/path/some/path`

File: fs_gcsfs/_gcsfs.py [hunk 4]
```diff
@@ -32,9 +33,11 @@ class GCSFS(FS):
     Args:
         bucket_name: The GCS bucket name.
         root_path: The root directory within the GCS Bucket
+        create: Whether to create ``root_path`` on initialization or not. If ``root_path`` does not yet exist and ``create=False`` an ``CreateFailed``
```
Comment (mathiaseitz) [inline fs_gcsfs/_gcsfs.py]: Minimal suggestion: I think it should be rather "... **a** CreateFailed exception" than "... **an** CreateFailed exception".

File: fs_gcsfs/_gcsfs.py [hunk 5]
```diff
@@ -66,8 +71,20 @@ def __init__(self,
         if self.client is None:
             self.client = Client()
 
-        self.bucket = self.client.get_bucket(self._bucket_name)
-        super(GCSFS, self).__init__()
+        try:
+            self.bucket = self.client.get_bucket(self._bucket_name)
+        except google.api_core.exceptions.NotFound as err:
+            raise CreateFailed("The provided bucket does not seem to exist") from err
```
Comment (mathiaseitz) [inline fs_gcsfs/_gcsfs.py]: Minimal suggestion: include the bucket's name in the Exception message.

File: fs_gcsfs/_gcsfs.py [hunk 6]
```diff
@@ -66,8 +71,20 @@ def __init__(self,
...
+        except google.api_core.exceptions.Forbidden as err:
+            raise CreateFailed("You don't have access to the provided bucket") from err
+
+        if create:
+            root_marker = self._get_blob(forcedir(root_path))
+            if not root_marker:
+                blob = self.bucket.blob(forcedir(root_path))
+                blob.upload_from_string(b"")
+        elif strict and not self._get_blob(forcedir(root_path)):
+            raise errors.CreateFailed("Root path does not exist")
```
Comment (mathiaseitz) [inline fs_gcsfs/_gcsfs.py]: Suggestion: include the actual value of root_path in the Exception message.

**Pull request merged**